acme.go

```diff
@@ -11,24 +12,32 @@ import (
 	"9fans.net/go/acme"
 )
 
+var nl = []byte("\n")
+
 // We would use io.Copy except for a bug in acme
 // where it crashes when reading trying to read more
 // than the negotiated 9P message size.
-func copyBody(w io.Writer, win *acme.Win) error {
+func copyBody(w io.Writer, win *acme.Win) (endsWithNL bool, err error) {
 	buf := make([]byte, 8000)
 	for {
 		n, err := win.Read("body", buf)
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
-			return err
+			return false, err
 		}
-		if _, err := w.Write(buf[0:n]); err != nil {
-			return fmt.Errorf("write error: %v", err)
+		if _, err := w.Write(buf[:n]); err != nil {
+			return false, fmt.Errorf("write error: %v", err)
 		}
+		endsWithNL = bytes.HasSuffix(buf[:n], nl)
 	}
-	return nil
+	if !endsWithNL {
+		if _, err := w.Write(nl); err != nil {
```

Comment:
Hmm, I like the idea but I don't think this is quite right. Consider, for example, a body that consists just of the single character "a". Running `apipe sha256sum` should result in the body being replaced by `sha256("a")` but AFAICS this change means it'll be replaced by `sha256("a\n")`.

I think that the only "right" solution here is to cope with the `\ No newline at end of file` line produced by diff in this case.

Reply:
You're right. I attempted to cope with `\ No newline at end of file` before but gave up. This approach was the one I was able to get done. It suits my usecase well, but you're right it's not a good general solution.